sdk/nodejs/output.ts

```diff
@@ -312,6 +236,73 @@ To manipulate the value of this Output, use '.apply' instead.`);
             },
         });
     }
+
+    public get(): T {
+        throw new Error(`Cannot call '.get' during update or preview.
+To manipulate the value of this Output, use '.apply' instead.`);
+    }
+
+    public apply<U>(func: (t: T) => Input<U>): Output<U> {
+        const applied = Promise.all([this.promise(), this.isKnown, this.isSecret])
+                               .then(([value, isKnown, isSecret]) => applyHelperAsync<T, U>(value, isKnown, isSecret, func));
+
+        const result = new OutputImpl<U>(
+            this.resources(), applied.then(a => a.value), applied.then(a => a.isKnown), applied.then(a => a.isSecret));
```

Comment:
note: if `applyHelperAsync` throws, then `applied` will be a rejected promise.  And these `.then` calls will just produce rejected promises.  That is the *right* behavior that we want IMO.

Reply:
> If we want to preserve the old behavior for those two bits, we can. But i would personally prefer not to.

How would this affect downlevel SDKs that attempt to serialize new outputs?

Reply:
>How would this affect downlevel SDKs that attempt to serialize new outputs?

They *may* get an exception at an earlier place than before.  i.e. at a point when some code awaits the 'isKnown' bit rather than the code awaiting the 'value' of the Output.

However, to be clear, that exception isn't something i believe ever actually happens in the wild.  Specifically, this code was written to be very paranoid about leaving Outputs in a bad state (i.e. with promises that never resolved, which could lead to hangs in apps).  The previous code simply made us be in the stated `(isUnknown: false, value: rejected)` the new code puts us in the state `(isKnown: rejected, value: rejected)`.  As any update would still have to get the value in either event, you would always end up throwing no matter what.